src/structures/CategoryChannel.js

```diff
@@ -7,6 +7,14 @@ const GuildChannel = require('./GuildChannel');
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
+  _patch(data) {
+    super._patch(data);
+
+    // Category Channels cannot have invites
+    this.createInvite = undefined;
+    this.fetchInvites = undefined;
```

Comment:
typings need updating for this, and perhaps this should be done in the constructor, and not every time the channel is updated

would it perhaps be an idea to use an interface (similar to `TextBasedChannel`) to apply these methods in the first place? idk

Reply:
Gotcha, I took inspiration for that from NewsChannel, which does it in the _patch method